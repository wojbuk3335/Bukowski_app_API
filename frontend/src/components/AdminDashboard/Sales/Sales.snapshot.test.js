import React from 'react';
import renderer from 'react-test-renderer';
import axios from 'axios';
import Sales from './Sales';

// Mock axios
jest.mock('axios');
const mockedAxios = axios;

// Mock all external dependencies
jest.mock('react-datepicker', () => {
    const mockReact = require('react');
    return {
        __esModule: true,
        default: function MockDatePicker(props) {
            return mockReact.createElement('input', {
                'data-testid': 'mock-datepicker',
                type: 'date',
                placeholder: props.placeholderText
            });
        },
        registerLocale: jest.fn()
    };
});

jest.mock('react-csv', () => ({
    CSVLink: ({ children, ...props }) => {
        const mockReact = require('react');
        return mockReact.createElement('button', { 'data-testid': 'csv-link', ...props }, children);
    }
}));

jest.mock('jspdf', () => {
    return jest.fn().mockImplementation(() => ({
        text: jest.fn(),
        addPage: jest.fn(),
        save: jest.fn()
    }));
});

jest.mock('jspdf-autotable', () => ({
    __esModule: true,
    default: jest.fn()
}));

jest.mock('xlsx', () => ({
    utils: {
        json_to_sheet: jest.fn(),
        book_new: jest.fn(),
        book_append_sheet: jest.fn()
    },
    writeFile: jest.fn()
}));

describe('Sales Component Snapshots', () => {
    beforeEach(() => {
        mockedAxios.get.mockClear();
    });

    test('should match snapshot with payment data', () => {
        const mockSalesData = [
            {
                _id: '1',
                fullName: 'IR 3212.313 BR√ÅZOWY',
                date: '2025-10-05',
                sellingPoint: 'Tata',
                size: '-',
                barcode: '0000401003136',
                timestamp: '2025-10-05T07:53:51.205Z',
                cash: [{ price: 250, currency: 'PLN' }],
                card: []
            },
            {
                _id: '2',
                fullName: 'Test Product',
                date: '2025-10-04',
                sellingPoint: 'Store A',
                size: 'M',
                barcode: '1234567890',
                timestamp: '2025-10-04T10:00:00Z',
                cash: [{ price: 100, currency: 'PLN' }],
                card: [{ price: 50, currency: 'PLN' }]
            }
        ];

        mockedAxios.get.mockResolvedValue({ data: mockSalesData });

        const tree = renderer.create(<Sales />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should match snapshot with no payment data', () => {
        const mockSalesData = [
            {
                _id: '3',
                fullName: 'No Payment Item',
                date: '2025-10-05',
                sellingPoint: 'Store B',
                size: 'L',
                barcode: '9876543210',
                timestamp: '2025-10-05T12:00:00Z',
                cash: [],
                card: []
            }
        ];

        mockedAxios.get.mockResolvedValue({ data: mockSalesData });

        const tree = renderer.create(<Sales />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should match snapshot with empty sales data', () => {
        mockedAxios.get.mockResolvedValue({ data: [] });

        const tree = renderer.create(<Sales />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should match snapshot with error state', () => {
        mockedAxios.get.mockRejectedValue(new Error('API Error'));

        const tree = renderer.create(<Sales />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should match snapshot with multi-currency payments', () => {
        const mockSalesData = [
            {
                _id: '4',
                fullName: 'Multi Currency Item',
                date: '2025-10-05',
                sellingPoint: 'International Store',
                size: 'XL',
                barcode: '5555666677',
                timestamp: '2025-10-05T14:00:00Z',
                cash: [
                    { price: 100, currency: 'EUR' },
                    { price: 50, currency: 'USD' }
                ],
                card: [
                    { price: 25, currency: 'GBP' }
                ]
            }
        ];

        mockedAxios.get.mockResolvedValue({ data: mockSalesData });

        const tree = renderer.create(<Sales />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});