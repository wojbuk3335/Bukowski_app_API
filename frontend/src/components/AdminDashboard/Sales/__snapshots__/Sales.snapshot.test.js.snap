// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sales Component Snapshots should match snapshot with empty sales data 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center mb-3"
  >
    <button
      className="btn btn-primary me-2"
      data={Array []}
      data-testid="csv-link"
      filename="sprzedaz.csv"
    >
      Eksportuj CSV
    </button>
    <button
      className="btn btn-warning me-2"
      onClick={[Function]}
    >
      Eksportuj PDF (Dane + Podsumowanie)
    </button>
    <button
      className="btn btn-success me-2"
      onClick={[Function]}
    >
      Drukuj Raport
    </button>
    <button
      className="btn btn-info me-3"
      onClick={[Function]}
    >
      Eksportuj Excel
    </button>
    <div
      className="d-flex align-items-center date-picker-container"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="me-2"
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Wyszukaj..."
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
        style={
          Object {
            "minWidth": "40px",
          }
        }
        title="Wybierz zakres dat "
      >
        üìÖ
      </button>
    </div>
  </div>
  <div
    className="tableContainer"
    id="sales-report-content"
  >
    <table
      className="table table responsiveTable text-center"
    >
      <caption
        className="caption"
      >
        Tabela przedstawiajƒÖca dane sprzeda≈ºy w systemie
      </caption>
      <thead>
        <tr>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Lp. 
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Pe≈Çna nazwa 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Data 
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Punkt sprzeda≈ºy 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Rozmiar 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Kod kreskowy 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            P≈Çatno≈õci 
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            Brak danych sprzeda≈ºy
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            <strong>
              ≈ÅƒÖczne przychody: 
            </strong>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`Sales Component Snapshots should match snapshot with error state 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center mb-3"
  >
    <button
      className="btn btn-primary me-2"
      data={Array []}
      data-testid="csv-link"
      filename="sprzedaz.csv"
    >
      Eksportuj CSV
    </button>
    <button
      className="btn btn-warning me-2"
      onClick={[Function]}
    >
      Eksportuj PDF (Dane + Podsumowanie)
    </button>
    <button
      className="btn btn-success me-2"
      onClick={[Function]}
    >
      Drukuj Raport
    </button>
    <button
      className="btn btn-info me-3"
      onClick={[Function]}
    >
      Eksportuj Excel
    </button>
    <div
      className="d-flex align-items-center date-picker-container"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="me-2"
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Wyszukaj..."
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
        style={
          Object {
            "minWidth": "40px",
          }
        }
        title="Wybierz zakres dat "
      >
        üìÖ
      </button>
    </div>
  </div>
  <div
    className="tableContainer"
    id="sales-report-content"
  >
    <table
      className="table table responsiveTable text-center"
    >
      <caption
        className="caption"
      >
        Tabela przedstawiajƒÖca dane sprzeda≈ºy w systemie
      </caption>
      <thead>
        <tr>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Lp. 
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Pe≈Çna nazwa 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Data 
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Punkt sprzeda≈ºy 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Rozmiar 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Kod kreskowy 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            P≈Çatno≈õci 
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            Brak danych sprzeda≈ºy
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            <strong>
              ≈ÅƒÖczne przychody: 
            </strong>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`Sales Component Snapshots should match snapshot with multi-currency payments 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center mb-3"
  >
    <button
      className="btn btn-primary me-2"
      data={Array []}
      data-testid="csv-link"
      filename="sprzedaz.csv"
    >
      Eksportuj CSV
    </button>
    <button
      className="btn btn-warning me-2"
      onClick={[Function]}
    >
      Eksportuj PDF (Dane + Podsumowanie)
    </button>
    <button
      className="btn btn-success me-2"
      onClick={[Function]}
    >
      Drukuj Raport
    </button>
    <button
      className="btn btn-info me-3"
      onClick={[Function]}
    >
      Eksportuj Excel
    </button>
    <div
      className="d-flex align-items-center date-picker-container"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="me-2"
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Wyszukaj..."
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
        style={
          Object {
            "minWidth": "40px",
          }
        }
        title="Wybierz zakres dat "
      >
        üìÖ
      </button>
    </div>
  </div>
  <div
    className="tableContainer"
    id="sales-report-content"
  >
    <table
      className="table table responsiveTable text-center"
    >
      <caption
        className="caption"
      >
        Tabela przedstawiajƒÖca dane sprzeda≈ºy w systemie
      </caption>
      <thead>
        <tr>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Lp. 
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Pe≈Çna nazwa 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Data 
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Punkt sprzeda≈ºy 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Rozmiar 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Kod kreskowy 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            P≈Çatno≈õci 
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            Brak danych sprzeda≈ºy
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            <strong>
              ≈ÅƒÖczne przychody: 
            </strong>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`Sales Component Snapshots should match snapshot with no payment data 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center mb-3"
  >
    <button
      className="btn btn-primary me-2"
      data={Array []}
      data-testid="csv-link"
      filename="sprzedaz.csv"
    >
      Eksportuj CSV
    </button>
    <button
      className="btn btn-warning me-2"
      onClick={[Function]}
    >
      Eksportuj PDF (Dane + Podsumowanie)
    </button>
    <button
      className="btn btn-success me-2"
      onClick={[Function]}
    >
      Drukuj Raport
    </button>
    <button
      className="btn btn-info me-3"
      onClick={[Function]}
    >
      Eksportuj Excel
    </button>
    <div
      className="d-flex align-items-center date-picker-container"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="me-2"
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Wyszukaj..."
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
        style={
          Object {
            "minWidth": "40px",
          }
        }
        title="Wybierz zakres dat "
      >
        üìÖ
      </button>
    </div>
  </div>
  <div
    className="tableContainer"
    id="sales-report-content"
  >
    <table
      className="table table responsiveTable text-center"
    >
      <caption
        className="caption"
      >
        Tabela przedstawiajƒÖca dane sprzeda≈ºy w systemie
      </caption>
      <thead>
        <tr>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Lp. 
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Pe≈Çna nazwa 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Data 
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Punkt sprzeda≈ºy 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Rozmiar 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Kod kreskowy 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            P≈Çatno≈õci 
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            Brak danych sprzeda≈ºy
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            <strong>
              ≈ÅƒÖczne przychody: 
            </strong>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;

exports[`Sales Component Snapshots should match snapshot with payment data 1`] = `
<div>
  <div
    className="d-flex justify-content-center align-items-center mb-3"
  >
    <button
      className="btn btn-primary me-2"
      data={Array []}
      data-testid="csv-link"
      filename="sprzedaz.csv"
    >
      Eksportuj CSV
    </button>
    <button
      className="btn btn-warning me-2"
      onClick={[Function]}
    >
      Eksportuj PDF (Dane + Podsumowanie)
    </button>
    <button
      className="btn btn-success me-2"
      onClick={[Function]}
    >
      Drukuj Raport
    </button>
    <button
      className="btn btn-info me-3"
      onClick={[Function]}
    >
      Eksportuj Excel
    </button>
    <div
      className="d-flex align-items-center date-picker-container"
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <div
        className="me-2"
        style={
          Object {
            "width": "200px",
          }
        }
      >
        <input
          className="form-control"
          onChange={[Function]}
          placeholder="Wyszukaj..."
          type="text"
          value=""
        />
      </div>
      <button
        className="btn btn-outline-secondary"
        onClick={[Function]}
        style={
          Object {
            "minWidth": "40px",
          }
        }
        title="Wybierz zakres dat "
      >
        üìÖ
      </button>
    </div>
  </div>
  <div
    className="tableContainer"
    id="sales-report-content"
  >
    <table
      className="table table responsiveTable text-center"
    >
      <caption
        className="caption"
      >
        Tabela przedstawiajƒÖca dane sprzeda≈ºy w systemie
      </caption>
      <thead>
        <tr>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Lp. 
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Pe≈Çna nazwa 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            Data 
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Punkt sprzeda≈ºy 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
          >
            <span
              onClick={[Function]}
              style={
                Object {
                  "cursor": "pointer",
                }
              }
            >
              Rozmiar 
            </span>
            <select
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              value=""
            >
              <option
                value=""
              >
                Wszystkie
              </option>
            </select>
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
          >
            Kod kreskowy 
            <input
              className="form-control form-control-sm mt-1"
              onChange={[Function]}
              placeholder="Filter"
              type="text"
            />
          </th>
          <th
            className="tableHeader noWrap"
            onClick={[Function]}
            style={
              Object {
                "textAlign": "center",
                "verticalAlign": "middle",
              }
            }
          >
            P≈Çatno≈õci 
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            Brak danych sprzeda≈ºy
          </td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td
            className="text-center"
            colSpan="7"
          >
            <strong>
              ≈ÅƒÖczne przychody: 
            </strong>
          </td>
        </tr>
      </tfoot>
    </table>
  </div>
</div>
`;
